kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/ScanSuite.scala

```diff
@@ -1594,6 +1594,38 @@ class ScanSuite extends AnyFunSuite with TestUtils with ExpressionTestUtils with
       )
     }
   }
+
+  test("read scan files with variant") {
+    val path = getTestResourceFilePath("spark-variant-delta")
+    val snapshot = Table.forPath(defaultEngine, path).getLatestSnapshot(defaultEngine)
+    val snapshotSchema = snapshot.getSchema(defaultEngine)
+
+    val expectedSchema = new StructType()
+      .add("id", LongType.LONG, true)
+      .add("v", VariantType.VARIANT, true)
+      .add("array_of_variants", new ArrayType(VariantType.VARIANT, true), true)
+      .add("struct_of_variants", new StructType().add("v", VariantType.VARIANT, true))
+      .add("map_of_variants", new MapType(StringType.STRING, VariantType.VARIANT, true), true)
+      .add(
+        "array_of_struct_of_variants",
+        new ArrayType(new StructType().add("v", VariantType.VARIANT, true), true),
+        true
+      )
+      .add(
+        "struct_of_array_of_variants",
+        new StructType().add("v", new ArrayType(VariantType.VARIANT, true), true),
+        true
+      )
+      .add("partitionKey", LongType.LONG, true)
+
+    assert(snapshotSchema == expectedSchema)
+
+    val scan = snapshot.getScanBuilder(defaultEngine).build()
```

Comment:
can you add one more test case the scan builder takes a predicate (on non-variant columns)? we just don't want the Kernel to error out. Predicates are best effort.

Reply:
Also can we add one more test case golden table that has checkpoint?